BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BluetoothBase.java

```diff
@@ -35,53 +31,58 @@ public BluetoothBase(Context Context, BluetoothStatusChanged handler) {
 
         //bluetooth = new BluetoothAdapter(this);
         bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null) {
+        if (bluetooth != null) {
```

Comment:
BTW, just to be fair, there is a pretty strong school of thought that argues that constructors shouldn't throw. It makes declaring variables tricky and can be a bit surprising since typically constructors shouldn't do anything. Instead someone should have to call a method on the object before anything interesting happens. I have a lot of sympathy for this later view. But in that case Start should definitely throw if bluetooth is null.